feat(docs): Add warning to new shield docs how Kconfig treats whitspaces

huber-th opened the pull request:
Resolves #2183

Added a warning to the shield section explaining that Kconfig does not ignore whitespaces on function calls and therefore adding whitespaces after the comma will break functionality.

caksoylar reviewed (approved):
That seems perfect, thank you!

caksoylar commented:
Unfortunately eslint seems to be unhappy, given the message I assume it might be related to the leading space example? I can't dig into it right now since I am traveling, but if you figure out a way to resolve it it would be great.

huber-th commented:
I fixed the issue. I switched to using `<code></code>` instead of backticks for the example in order to being able to use HTML tag for the whitespace instead of a non-regular whitespace. Markdown trims leading and trailing whitespaces by default.

I added a second commit updating the `CONTRIBUTING.md` to mention the usage of ESLint for changes to the documentation.

Let me know in case that shouldn't be done as part of this PR but rather separately.

caksoylar commented:
Thanks, the new commit is fine in this PR IMO. I have to admit I have been forgetting to update contributing.md, but linting is documented in https://zmk.dev/docs/development/documentation#formatting-and-linting-your-changes.
